src/postgres/commands/createPostgresServer/steps/PostgresServerNameStep.ts

```diff
@@ -46,8 +47,7 @@ async function validatePostgresServerName(name: string, client: PostgreSQLManage
         return localize('serverNamePrefixSuffixCheck', 'Server name must not start or end in a hyphen.');
     }
 
-    const availabilityRequest: NameAvailabilityRequest = { name: name, type: "Microsoft.DBforPostgreSQL" };
-    const availability: NameAvailability = (await client.checkNameAvailability.execute(availabilityRequest));
+    const availability: AbstractNameAvailability = await client.checkNameAvailability(serverType, name);
 
     if (!availability.nameAvailable) {
         if (availability.reason === 'AlreadyExists') {
```

Comment:
If I follow my suggested pattern for `type` vs. `interface`, it'll look something like this:
```typescript
export type AbstractNameAvailability = SingleModels.NameAvailability | FlexibleModels.NameAvailability;
```

However, all of the sudden this line starts complaining because `reason` doesn't exist on `FlexibleModels.NameAvailability`. That implies to me that either:
1. The types on `FlexibleModels.NameAvailability` are wrong
2. The types are correct and this logic doesn't actually work for flexible servers

If 2 is true, we need to update this logic. If 1 is true, we can update the type like so:
```typescript
export type AbstractNameAvailability = SingleModels.NameAvailability | (FlexibleModels.NameAvailability & {
    /**
    * NOTE: The types on `FlexibleModels.NameAvailability` are incorrect, this property does exist
    */
    reason?: string;
})
```

Reply:
thats just because I referred to the reason instead of the message, this can be unionised.